mdgo/util.py

```diff
@@ -433,53 +435,81 @@ def ff_parser(ff_dir, xyz_dir):
+            same length as solvents. For example, for a 30% / 70% (w/w) mix of
+            two solvent, pass [0.3, 0.7] or [30, 70]. The sum of weights / volumes
+            does not need to be normalized.
+        num_salt: The number of salt in the box.
+        mode: Weight mode (Wweight/weight/W/w/W./w.) or volume mode
+            (Volume/volume/V/v/V./v.) for interpreting the ratio of solvents.
+        radii_type: "Bondi", "Lange", or "pymatgen". Bondi and Lange vdW radii
+            are compiled in this package for H, B, C, N, O, F, Si, P, S, Cl, Br,
+            and I. Choose 'pymatgen' to use the vdW radii from pymatgen.Element,
+            which are available for most elements and reflect the latest values in
+            the CRC handbook.
 
     Returns:
         (list, float):
-            A list of molecule numbers in the box starting with the salt,
-            and the approximate box length
+            A list the number of molecules in the simulation box, starting with
+            the salt and followed by each solvent in 'solvents'. The list is followed
+            by a float of the approximate length of one side of the box in Å.
 
     """
     n_solvent = list()
     n = len(solv_ratio)
-    if isinstance(salt, numbers.Number):
+    if n != len(solvents):
+        raise ValueError("solvents and solv_ratio must be the same length!")
+
+    if isinstance(salt, float) or isinstance(salt, int):
         salt_molar_volume = salt
-    elif type(salt) is Molecule:
-        salt_molar_volume = molecular_volume(salt, salt.composition.reduced_formula)
-    elif salt.lower() in MOLAR_VOLUME:
-        salt_molar_volume = MOLAR_VOLUME.get(salt.lower())
-    else:
-        if not os.path.exists(salt):
-            print("\nError: Input file '{}' not found.\n".format(salt))
-            sys.exit(1)
-        name = os.path.splitext(os.path.split(salt)[-1])[0]
-        ext = os.path.splitext(os.path.split(salt)[-1])[1]
-        if not ext == ".xyz":
-            print("Error: Wrong file format, please use a .xyz file.\n")
-            sys.exit(1)
-        salt_molar_volume = molecular_volume(salt, name)
+    elif isinstance(salt, Molecule):
+        salt_molar_volume = molecular_volume(salt, salt.composition.reduced_formula, radii_type=radii_type)
+    elif isinstance(salt, str):
+        if MOLAR_VOLUME.get(salt.lower()):
+            salt_molar_volume = MOLAR_VOLUME.get(salt.lower(), 0)
+        else:
+            if not os.path.exists(salt):
+                print("\nError: Input file '{}' not found.\n".format(salt))
+                sys.exit(1)
+            name = os.path.splitext(os.path.split(salt)[-1])[0]
+            ext = os.path.splitext(os.path.split(salt)[-1])[1]
+            if not ext == ".xyz":
+                print("Error: Wrong file format, please use a .xyz file.\n")
+                sys.exit(1)
+            salt_molar_volume = molecular_volume(salt, name, radii_type=radii_type)
```

Comment:
Does python do strict type checking? If not, then should we catch the invalid `salt` type error?

Reply:
I added an extra `else` to raise `ValueError` if salt is not once of the valid types. Is that what you mean?